src/wrappers/strategies/AssetPairManagerWrapper.ts

```diff
@@ -0,0 +1,192 @@
+   * manager by checking that the time and price constraints have been met then logging a timestamp used later in the
+   * process to confirm the signal.
+   *
+   * @param  managerAddress               Address of the rebalancing manager contract
+   * @return                              The hash of the resulting transaction.
+   */
+  public async initialPropose(
+    managerAddress: Address,
+    txOpts?: Tx,
+  ): Promise<string> {
+    const assetPairManagerInstance = await this.contracts.loadAssetPairManagerContractAsync(managerAddress);
+    const txOptions = await generateTxOpts(this.web3, txOpts);
+
+    return await assetPairManagerInstance.initialPropose.sendTransactionAsync(txOptions);
+  }
+
+  /**
+   * Calls an AssetPairManager's confirmPropose function. This function again checks to make sure that
+   * price and time constraints have been satisfied. After that it generates the new allocation proposal and sends the
+   * results to the Rebalancing Set Token to kick off the official rebalance.
+   *
+   * @param  managerAddress               Address of the rebalancing manager contract
+   * @return                              The hash of the resulting transaction.
+   */
+  public async confirmPropose(
+    managerAddress: Address,
+    txOpts?: Tx,
+  ): Promise<string> {
+    const assetPairManagerInstance = await this.contracts.loadAssetPairManagerContractAsync(managerAddress);
+    const txOptions = await generateTxOpts(this.web3, txOpts);
+
+    return await assetPairManagerInstance.confirmPropose.sendTransactionAsync(txOptions);
+  }
+
+  /**
+   * Calls an AssetPairManager's canInitialPropose function. Returns whether initialPropose can be called
+   * without reverting.
+   *
+   * @param  managerAddress               Address of the rebalancing manager contract
+   * @return                              Boolean indicating whether initialPropose can be successfully called
+   */
+  public async canInitialPropose(managerAddress: Address): Promise<boolean> {
+    const assetPairManagerInstance = await this.contracts.loadAssetPairManagerContractAsync(managerAddress);
+
+    return await assetPairManagerInstance.canInitialPropose.callAsync()
+      .then(value => { return value; })
+      .catch(error => { return false; });
+  }
+
+  /**
+   * Calls an AssetPairManager's canConfirmPropose function. Returns whether confirmPropose can be called
+   * without reverting.
+   *
+   * @param  managerAddress               Address of the rebalancing manager contract
+   * @return                              Boolean indicating whether confirmPropose can be successfully called
+   */
+  public async canConfirmPropose(managerAddress: Address): Promise<boolean> {
+    const assetPairManagerInstance = await this.contracts.loadAssetPairManagerContractAsync(managerAddress);
+
+    return await assetPairManagerInstance.canConfirmPropose.callAsync()
+      .then(value => { return value; })
```

Comment:
I think this logic should exist on the API layer. This should throw like the contract would in the event of a revert. Is this possible? If not, leave like this for now.